### ISSUE TITLE
<Transition> component crashes on react 16.4.X when using 'keys' prop

### ISSUE THREAD

**Issue by Tarpsvo** [drcmda/react-spring#161]
The `<Transition>` component crashes when using React 16.4.X with `keys` prop, while it works as intended on React 16.3.X.

**The "2-state and 1-state Reveals" demo shown in README using React 16.3.1:**
It works with and without `keys` prop.

Link: https://codesandbox.io/embed/yj52v5689

**The same demo using React 16.4.1:**
Without `keys` prop: no animation.
With `keys` prop: animates first one and then crashes.

Link: https://codesandbox.io/embed/03zyjm05vp

----------

Apparently `getDerivedStateFromProps` is returning `transitions` array with one value as `undefined`. I don't currently grasp the internals of the `getDerivedStateFromProps` function and am clueless as to why this issue happens. Especially in React 16.4. I do know that React 16.4 did change the behaviour of `getDerivedStateFromProps`. In 16.3, it used to fire only when props AND state change, but as of 16.4.0, it fires if props OR state changes. I do not really see why this would be a breaking change for `react-spring` though.

**Comment by Tarpsvo** [comment]
After a lot of debugging, I found the cause and created a working fix. It seems it was related to the case where keys are numbers, not strings.

Looking at the code:
`let currentKeys = Object.keys(current)` creates an array of strings (ie ["0"]).

Later, when filtering them:
`let added = keys.filter(item => !currentSet.has(item))` it searched for `new Set([0]).has("0")`, which obviously turns out to be false.

My quick-fix was to just compare all `keys` as strings.

If it's not a proper solution, I'm totally fine with that. I just hope I helped you get closer to a proper solution.

I am still dumbfounded as to why React 16.4.X crashed and 16.3.X did not.

Thanks for the great work on this library!
